presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeRemoteFunctions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createLineitem;
+
+public abstract class AbstractTestNativeRemoteFunctions
+        extends AbstractTestQueryFramework
+{
+    public String remoteFunctionServerUds;
+    public String remoteFunctionServerBinaryPath;
+
+    @Override
+    protected void preInitQueryRunners()
+    {
+        // Initialize the remote function thrift server process.
+        remoteFunctionServerBinaryPath = System.getProperty("REMOTE_FUNCTION_SERVER");
+        if (remoteFunctionServerBinaryPath != null) {
```

Comment:
Why would we want to skip the test? Can this mask issues? Let's require that this property is set and fail the test it isn't.

Reply:
@mbasmanova happy to make the change, but the tricky part here is that the remote function binary not be available, for example, if the PRESTO_ENABLE_REMOTE_FUNCTIONS build flag was not specified. In that case, just running all unit tests would result in failures.

Reply:
I suggest we introduce a new maven profile (using `maven-surefire-plugin` groups) for remote function tests. We do this for specialised tests like hive, parquet, and other meta internal integrations.

If we want to disable remote function tests by default, we could add:
`<excludeGroups>remote-functions</exludeGroups>`
to 
https://github.com/prestodb/presto/blob/6465ee8a6b57e653606dca6ca433db54e3f1dc53/presto-native-execution/pom.xml#L191


And introduce a new test suite exclusively for remote functions, like this:
```
          <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <configuration>
                    <forkCount>1</forkCount>
                    <reuseForks>false</reuseForks>
                    <groups>remote-functions</groups>
                    <systemPropertyVariables>
                        <PRESTO_SERVER>/root/project/build/debug/presto_cpp/main/presto_server</PRESTO_SERVER>
                        <DATA_DIR>/tmp/velox</DATA_DIR>
                    </systemPropertyVariables>
                </configuration>
            </plugin>
```

With this change we could then mark `REMOTE_FUNCTION_SERVER` as a require property in `preInitQueryRunners`.

Reply:
Nice! Let me look into this.

Reply:
Something like this: https://github.com/prestodb/presto/pull/20658/commits/6868a3364589a8f38b8cd18dd0c0e02a6d78afda

Reply:
thanks @vermapratyush. I incorporated your changes. Could you take one last look?